operator/controllers/utils.go

```diff
@@ -607,6 +616,35 @@ func CheckAndGenerateCertForOperator() error {
 		return err
 	}
 
+	secretName := fmt.Sprintf("%s-cert", serviceName)
+	secret := &corev1.Secret{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+			Labels:    CommonLabels(map[string]string{}, serviceName, AppName),
+		},
+	}
+	err = k8sClient.Get(context.Background(), types.NamespacedName{Name: secretName, Namespace: namespace}, secret)
+	if err != nil {
+		secret.Data = map[string][]byte{
+			OperatorCertName:    serverCertPEM.Bytes(),
+			OperatorCertKeyName: serverPrivKeyPEM.Bytes(),
+			OperatorCAName:      caPEM.Bytes(),
+		}
+		_, err = controllerutil.CreateOrUpdate(context.Background(), k8sClient, secret, SecretMutate(secret, secret.Data, secret.OwnerReferences))
+		if err != nil {
+			return err
+		}
+	}
+
+	block, _ := pem.Decode(secret.Data[OperatorCertName])
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err == nil && cert.NotAfter.After(time.Now()) {
```

Comment:
What if err is not nil?

Reply:
If error is not nil, the new certificates which are generated above will be used instead

Reply:
If error is not nil, the new certificates which are generated above will be used instead

Reply:
I see a potential issue in this and fixing it. Thank you for pointing this out